.github/workflows/update_learn_playlists.yml

```diff
@@ -5,16 +5,32 @@ on:
     inputs:
       mode:
         description: 'Mode: stable, testing'
+        default: 'testing'
         required: true
+  workflow_call:
+    inputs:
+      mode:
+        description: 'Mode: stable, testing'
         default: 'testing'
+        type: string
+        required: true
+      environment:
+        description: "Environment: use 'production' to prompt for approval"
+        default: ''
+        type: string
+        required: false
```

Comment:
Ideally we would do it in `deploy.yml` instead, like this:

```YML
  update_learn_playlists:
    name: 'Update Home→Learn playlists'
    uses: ./.github/workflows/update_learn_playlists.yml
    environment: production # Not allowed by GitHub
```

Since GitHub doesn't allow `environment` to be specified with `uses`, we have to pass it through as an `input` instead (i.e. via `with: environment: production`) and rely on the called workflow to do something with it (which it does [here](https://github.com/musescore/MuseScore/pull/24116/files#diff-21f49fdb8fec9cf9eee368a5b51985fef87c4d117334cd82ed116da5b01e565fR32)).

As you say, when triggered via `workflow_call`, the input type [has to be](https://docs.github.com/en/enterprise-cloud@latest/actions/writing-workflows/workflow-syntax-for-github-actions#onworkflow_callinputsinput_idtype) `string` rather than `environment`.